### PR TITLE
Windows: Refactor resources structure

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

This refactors the exec driver resources structure into platform specific parts.

### CODE REVIEW

**Comment by jhowardmsft** [comment]
Windows Jenkins failure is a well know current problem with that test.

**Comment by icecrime** [comment]
Ping @crosbymichael @LK4D4!

**Comment by LK4D4** [comment]
Design looks ok

**Comment by jhowardmsft** [comment]
@icecrime Does that mean this can move to code review?

**Comment by jhowardmsft** [comment]
Rebased.

**Comment by cpuguy83** [comment]
LGTM

**Comment by jhowardmsft** [comment]
Rebased

**Comment by thaJeztah** [comment]
@jhowardmsft needs another rebase :cry:

**Comment by jhowardmsft** [comment]
@thaJeztah Done. I was kind of expecting that - tripped up over the volumes PR ;)

File: daemon/container_windows.go [hunk 1]
```diff
@@ -86,7 +86,9 @@ func populateCommand(c *Container, env []string) error {
 
 	// TODO Windows. More resource controls to be implemented later.
 	resources := &execdriver.Resources{
-		CPUShares: c.hostConfig.CPUShares,
+		CommonResources: execdriver.CommonResources{
+			CPUShares: c.hostConfig.CPUShares,
```
Comment (LK4D4) [inline daemon/container_windows.go]: Why we set only `CPUShares` here?

Comment (jhowardmsft) [inline daemon/container_windows.go]: Because that's all Windows supports in TP4.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**